src/test/unit_tests/jest_tests/test_config_dir_restructure_upgrade_script.test.js

```diff
@@ -0,0 +1,654 @@
+        const account_upgrade_params = {
+            ...new_account_data, account_name: new_account_data.name, identity_path,
+            account_old_path,
+            account_old_path_stat, identity_dir_path
+        };
+        await create_identity_if_missing(default_config_fs.fs_context, account_upgrade_params, dbg);
+        await assert_identity_created(new_account_data);
+    });
+});
+
+describe('prepare_account_upgrade_params', () => {
+    beforeAll(async () => {
+        await fail_test_if_default_config_dir_exists('test_config_dir_restructure', default_config_fs);
+    });
+
+    beforeEach(async () => {
+        await create_config_dir(config.NSFS_NC_DEFAULT_CONF_DIR);
+        await default_config_fs.create_config_dirs_if_missing();
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        await create_fresh_path(hidden_old_accounts_path);
+    }, TEST_TIMEOUT);
+
+    afterEach(async () => {
+        await clean_config_dir(default_config_fs);
+    }, TEST_TIMEOUT);
+
+    it('prepare_account_upgrade_params() - account name exists', async () => {
+        const new_account_data = { _id: String(1), name: 'old_account' + 1, user: 'root', access_keys: [{ access_key: mock_access_key }] };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data);
+        const account_old_path = default_config_fs._get_old_account_path_by_name(new_account_data.name);
+        const identity_path = default_config_fs.get_identity_path_by_id(new_account_data._id);
+        const identity_dir_path = default_config_fs.get_identity_dir_path_by_id(new_account_data._id);
+        let src_file;
+        let actual_upgrade_params;
+        try {
+            src_file = await native_fs_utils.open_file(fs_context, undefined, account_old_path, 'r');
+            const account_old_path_stat = await src_file.stat(default_config_fs.fs_context);
+            const expected_account_upgrade_params = {
+                _id: new_account_data._id,
+                access_keys: new_account_data.access_keys,
+                account_name: new_account_data.name,
+                identity_path,
+                account_old_path,
+                account_old_path_stat,
+                identity_dir_path,
+                src_file,
+                gpfs_options: { src_file, dst_file: undefined }
+            };
+            actual_upgrade_params = await prepare_account_upgrade_params(default_config_fs, new_account_data.name);
+            assert_upgrade_params(actual_upgrade_params, expected_account_upgrade_params);
+        } catch (err) {
+            fail(err);
+        } finally {
+            if (src_file) await src_file.close(default_config_fs.fs_context);
+            if (actual_upgrade_params.src_file) await actual_upgrade_params.src_file.close(default_config_fs.fs_context);
+        }
+    });
+
+    it('prepare_account_upgrade_params() - account name exists', async () => {
```

Comment:
Are you sure about the test title here? (in the previous test, we had the same title.